File: Leetcode_solutions/House-Robber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var rob = function(nums) {
  let a = 0,
    b = 0;

  for (let i = 0; i < nums.length; i++) {
    if (i % 2 == 0) {
      a = Math.max(a + nums[i], b);
    } else {
      b = Math.max(a, b + nums[i]);
    }
  }

  return Math.max(a, b);
};

console.log(rob([2, 1, 1, 2]));
console.log(rob([1, 2, 1, 1, 3]));

File: Leetcode_solutions/House_Robber.py
class Solution:
    current = 0

    def rob(self, nums: List[int]) -> int:
        for i in range(len(nums)):
            first = nums[i] + nums[i - 2]
            second = max(nums[i - 2], nums[i - 1])
            current = max(first, second)
        return current


class Solution:
    def rob(self, nums: List[int]) -> int:
        prev = 0
        curr = 0
        for i in range(len(nums)):
            temp = curr
            curr = max(prev + nums[i], curr)
            prev = temp
        return curr
